src/main/java/org/jabref/gui/groups/GroupTreeView.java

```diff
@@ -120,27 +120,29 @@ public void initialize() {
                 .withTooltip(GroupNodeViewModel::getDescription)
                 .install(mainColumn);
 
-        // Number of hits
-        PseudoClass anySelected = PseudoClass.getPseudoClass("any-selected");
-        PseudoClass allSelected = PseudoClass.getPseudoClass("all-selected");
-        new ViewModelTreeTableCellFactory<GroupNodeViewModel>()
-                .withGraphic(group -> {
-                    final StackPane node = new StackPane();
-                    node.getStyleClass().setAll("hits");
-                    if (!group.isRoot()) {
-                        BindingsHelper.includePseudoClassWhen(node, anySelected,
-                                group.anySelectedEntriesMatchedProperty());
-                        BindingsHelper.includePseudoClassWhen(node, allSelected,
-                                group.allSelectedEntriesMatchedProperty());
-                    }
-                    Text text = new Text();
-                    text.textProperty().bind(group.getHits().asString());
-                    text.getStyleClass().setAll("text");
-                    node.getChildren().add(text);
-                    node.setMaxWidth(Control.USE_PREF_SIZE);
-                    return node;
-                })
-                .install(numberColumn);
+        // Number of hits (only if user wants to see them)
+        if (preferencesService.getDisplayGroupCount()) {
+            PseudoClass anySelected = PseudoClass.getPseudoClass("any-selected");
+            PseudoClass allSelected = PseudoClass.getPseudoClass("all-selected");
+            new ViewModelTreeTableCellFactory<GroupNodeViewModel>()
+                    .withGraphic(group -> {
+                        final StackPane node = new StackPane();
+                        node.getStyleClass().setAll("hits");
+                        if (!group.isRoot()) {
+                            BindingsHelper.includePseudoClassWhen(node, anySelected,
+                                    group.anySelectedEntriesMatchedProperty());
+                            BindingsHelper.includePseudoClassWhen(node, allSelected,
+                                    group.allSelectedEntriesMatchedProperty());
+                        }
+                        Text text = new Text();
```

Comment:
As noted in the issue, the circle node should always be displayed because it is also used for the `any/all selected` status. Can you thus please add the if check only around the text here.

Reply:
Done,
please see new GroupTreeeView.java. OK now?

Reply:
Thanks for quick follow-up! Looks good indeed.